modules/server/src/middlewares/xviz-websocket-sender.js

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import {XVIZFormat, XVIZFormatWriter} from '@xviz/io';
+
+export class WebsocketSink {
+  constructor(socket, options) {
+    this.socket = socket;
+    this.options = options;
+  }
+
+  writeSync(name, data) {
```

Comment:
I can move the name to the send argument, that way it can be ignored by JS impl's that don't require it.

Reply:
So this runs a bit deeper than I thought.  The writeSync i related to the other write messages (specifically writeMetadata and writeFrame).  Since we are changing the writeFrame => writeMessage I"ll make this related IO change in that PR